packages/desktop-gui/cypress/integration/ipc_bus_spec.js

```diff
@@ -0,0 +1,33 @@
+import ipcBus from '../../src/lib/ipc-bus'
+
+const RANDOM_NUMBER = 0.5
+
+describe('IPC bus', () => {
+  beforeEach(() => {
+    cy.stub(Math, 'random').returns(RANDOM_NUMBER)
+    cy.stub(window.ipc, 'on').returns()
+    cy.stub(window.ipc, 'send').returns()
+  })
+
+  it('sends event as expected', () => {
+    ipcBus('foo:bar', 'baz', 'quux')
+
+    expect(window.ipc.send).to.be.calledWith('request', RANDOM_NUMBER, 'foo:bar', 'baz', 'quux')
```

Comment:
My Zach, since you don't care about the value of the second argument, you can skip stabbing `Math.random` by using Sinon matchers. See https://docs.cypress.io/api/utilities/sinon.html#match-string and https://sinonjs.org/releases/latest/matchers/ and https://github.com/cypress-io/cypress-example-kitchensink/pull/385/files

Reply:
thanks for the links :) i know about matchers, i just preferred this since it asserts that it's a random number